src/components/Executions/filters/test/useOnlyMyExecutionsFilterState.test.ts

```diff
@@ -0,0 +1,33 @@
+import { loadedFetchable } from 'components/hooks/__mocks__/fetchableData';
+import { useUserProfile } from 'components/hooks/useUserProfile';
+import { useOnlyMyExecutionsFilterState } from 'components/Executions/filters/useOnlyMyExecutionsFilterState';
+import { UserProfile } from 'models/Common/types';
+import { FetchableData } from 'components/hooks/types';
+import { renderHook } from '@testing-library/react-hooks';
+
+jest.mock('components/hooks/useUserProfile');
+
+describe('useOnlyMyExecutionsFilterState', () => {
+  const mockUseRemoteLiteralMap = useUserProfile as jest.Mock<FetchableData<UserProfile | null>>;
+  mockUseRemoteLiteralMap.mockReturnValue(loadedFetchable(null, jest.fn()));
+
+  describe.each`
+    isFilterDisabled | initialValue | expected
+    ${undefined}     | ${undefined} | ${{ isFilterDisabled: false, onlyMyExecutionsValue: false }}
+    ${false}         | ${undefined} | ${{ isFilterDisabled: false, onlyMyExecutionsValue: false }}
+    ${true}          | ${undefined} | ${{ isFilterDisabled: true, onlyMyExecutionsValue: false }}
+    ${undefined}     | ${false}     | ${{ isFilterDisabled: false, onlyMyExecutionsValue: false }}
+    ${undefined}     | ${true}      | ${{ isFilterDisabled: false, onlyMyExecutionsValue: true }}
+    ${false}         | ${false}     | ${{ isFilterDisabled: false, onlyMyExecutionsValue: false }}
+    ${false}         | ${true}      | ${{ isFilterDisabled: false, onlyMyExecutionsValue: true }}
+    ${true}          | ${false}     | ${{ isFilterDisabled: true, onlyMyExecutionsValue: false }}
+    ${true}          | ${true}      | ${{ isFilterDisabled: true, onlyMyExecutionsValue: true }}
+  `('for each case', ({ isFilterDisabled, initialValue, expected }) => {
```

Comment:
Initially wrote this up as individual tests, but thought this may be a good use case to digest them as a table. I have the original code handy, so wouldn't be a problem to revert, if it's just my taste :)

Reply:
I don't have any particular taste, but curious how it will look like if one of them fails?

Reply:
@anrusina only failed cases will show up, like in this screenshot, test case title is customized on line 26, and I just stringified `expected` to show value instead of [Object Object]  
<img width="958" alt="Screen Shot 2022-03-21 at 5 42 53 PM" src="https://user-images.githubusercontent.com/101579322/159375195-4df968a2-d8ca-4375-9bd2-e5f4fb8aa738.png">

Reply:
Looks good 👍